src/patternfly/components/Select/select.scss

```
@@ -395,6 +403,24 @@
   &.pf-m-selected {
     --pf-c-select__menu-item--PaddingRight: var(--pf-c-select__menu-item--m-selected--PaddingRight);
   }
+
+  &.pf-m-description {
+    --pf-c-select__menu-item--m-selected--PaddingRight: 0;
```

and if you made the change above, this would change to 

```suggestion
    --pf-c-select__menu-item--PaddingRight: 0;
```